src/Symfony/Bundle/FrameworkBundle/Resources/config/cache.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Cache\Adapter\AbstractAdapter;
+use Symfony\Component\Cache\Adapter\AdapterInterface;
+use Symfony\Component\Cache\Adapter\ApcuAdapter;
+use Symfony\Component\Cache\Adapter\ArrayAdapter;
+use Symfony\Component\Cache\Adapter\DoctrineAdapter;
+use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+use Symfony\Component\Cache\Adapter\MemcachedAdapter;
+use Symfony\Component\Cache\Adapter\PdoAdapter;
+use Symfony\Component\Cache\Adapter\ProxyAdapter;
+use Symfony\Component\Cache\Adapter\RedisAdapter;
+use Symfony\Component\Cache\Adapter\TagAwareAdapter;
+use Symfony\Component\Cache\Marshaller\DefaultMarshaller;
+use Symfony\Component\HttpKernel\CacheClearer\Psr6CacheClearer;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\TagAwareCacheInterface;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('cache.app')
+            ->parent('cache.adapter.filesystem')
+            ->public()
+            ->tag('cache.pool', ['clearer' => 'cache.app_clearer'])
+
+        ->set('cache.app.taggable', TagAwareAdapter::class)
+            ->args([service('cache.app')])
+
+        ->set('cache.system')
+            ->parent('cache.adapter.system')
+            ->public()
+            ->tag('cache.pool')
+
+        ->set('cache.validator')
+            ->parent('cache.system')
+            ->tag('cache.pool')
+
+        ->set('cache.serializer')
+            ->parent('cache.system')
+            ->tag('cache.pool')
+
+        ->set('cache.annotations')
+            ->parent('cache.system')
+            ->tag('cache.pool')
+
+        ->set('cache.property_info')
+            ->parent('cache.system')
+            ->tag('cache.pool')
+
+        ->set('cache.messenger.restart_workers_signal')
+            ->parent('cache.app')
+            ->tag('cache.pool')
+
+        ->set('cache.adapter.system', AdapterInterface::class)
+            ->abstract()
+            ->factory([AbstractAdapter::class, 'createSystemCache'])
+            ->args([
+                '', // namespace
```

Comment:
I think we should use `abstract_arg` instead of '', @javiereguiluz ?

Reply:
I'm not sure about that ... let's ask @Tobion and @nicolas-grekas Thanks!

Reply:
that could be a BC break so better not

Reply:
well, it depends whether we always replace it (and also whether passing the empty string in the final service is a valid configuration)

Reply:
Replacing or not depends on how the child def is defined, we don't control that. The empty string is a valid namespace yes. So better as is.

Reply:
So, shall I resolve this conversation?

Reply:
It's fine to keep this as empty namespace. But in other cases like `'', // Memcached connection service` this should be an abstract_arg because an empty string is not a valid argument for the MemcachedAdapter.
So please just use abstract_arg where the default empty string does not make sense like dsn or connection objects or other services required to be injected.

Reply:
Updated